addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py

```diff
@@ -442,8 +442,8 @@ def _invoice_constraints_peppol_en16931_ubl(self, invoice, vals):
                 # KVK or OIN number (schemeID 0106 or 0190)
                 'nl_r_003': _(
                     "The supplier %s must have a KVK or OIN number.",
-                    vals['supplier'].display_name
-                ) if 'l10n_nl_oin' not in vals['supplier']._fields or 'l10n_nl_kvk' not in vals['supplier']._fields else '',
+                    vals['supplier'].commercial_partner_id.display_name
+                ) if 'l10n_nl_oin' not in vals['supplier'].commercial_partner_id._fields or 'l10n_nl_kvk' not in vals['supplier'].commercial_partner_id._fields else '',
```

Comment:
maybe use the value in `vals` (see: https://docs.peppol.eu/poacc/billing/3.0/rules/ubl-peppol/NL-R-003/) for the path